### ISSUE TITLE
Memory leaks: Picom leaks memory when using "geometry-change" animation 

### ISSUE THREAD

**Issue by Husto1** [yshui/compton#1383]
### Platform
Linux gentoo 6.6.47-gentoo

### GPU, drivers, and screen setup
Intel iGPU 11th Gen Intel(R) Core(TM) i5-11400H @ 2.70GHz, Mesa 24.1.7, libva-intel-media-driver 24.2.5

<details>
<summary>glxinfo -B</summary>

```
name of display: :0
display: :0  screen: 0
direct rendering: Yes
Extended renderer info (GLX_MESA_query_renderer):
    Vendor: Intel (0x8086)
    Device: Mesa Intel(R) UHD Graphics (TGL GT1) (0x9a68)
    Version: 24.1.7
    Accelerated: yes
    Video memory: 7891MB
    Unified memory: yes
    Preferred profile: core (0x1)
    Max core profile version: 4.6
    Max compat profile version: 4.6
    Max GLES1 profile version: 1.1
    Max GLES[23] profile version: 3.2
OpenGL vendor string: Intel
OpenGL renderer string: Mesa Intel(R) UHD Graphics (TGL GT1)
OpenGL core profile version string: 4.6 (Core Profile) Mesa 24.1.7
OpenGL core profile shading language version string: 4.60
OpenGL core profile context flags: (none)
OpenGL core profile profile mask: core profile

OpenGL version string: 4.6 (Compatibility Profile) Mesa 24.1.7
OpenGL shading language version string: 4.60
OpenGL context flags: (none)
OpenGL profile mask: compatibility profile

OpenGL ES profile version string: OpenGL ES 3.2 Mesa 24.1.7
OpenGL ES profile shading language version string: OpenGL ES GLSL ES 3.20
```
</details>

### Environment
dwm-6.5 with 
<details>
<summary>these patches</summary>

```
dwm-actualfullscreen-20211013-cb3f58a.diff  
dwm-functionalgaps-pertagfunctionality-6.2.diff
dwm-alwayscenter-20200625-f04cac6.diff      
dwm-movestack-20211115-a786211.diff
dwm-attachbottom-6.3.diff                   
dwm-pertag-20200914-61bb8b2.diff
dwm-dynamicswallow-20240320-061e9fe.diff
```
</details>

### picom version
v12.3

<details>
<summary>Diagnostics</summary>
**Version:** v12.3

### Extensions:

* Shape: Yes
* RandR: Yes
* Present: Present

### Misc:

* Use Overlay: No

* Config file specified: None
* Config file used: /home/husto/.config/picom.conf

### Drivers (inaccurate):

modesetting
[ 11/06/2024 22:27:00.543 egl_init WARN ] The egl backend is still experimental, use with care.

### Backend: egl

* Driver vendors:
 * EGL: Mesa Project
 * EGL driver: iris
 * GL: Intel
* GL renderer: Mesa Intel(R) UHD Graphics (TGL GT1)

### Backend: glx

* Driver vendors:
 * GLX: Mesa Project and SGI
 * GL: Intel
* GL renderer: Mesa Intel(R) UHD Graphics (TGL GT1)
* Accelerated: 1

</details>

### Configuration:
<details>
    <summary>Configuration file</summary>

```
backend = "glx"
animations = ({
	triggers = [ "geometry" ];
	preset = "geometry-change";
})
```
</details>

### Steps of reproduction
1. Run picom with glx backend
2. Add these to config file:
```
animations = ({
	triggers = [ "geometry" ];
	preset = "geometry-change";
})
```
3. Do stuff (e.g., resize the window) and watch how memory usage grows over time
4. Killing picom will instantly free up all memory
### Expected behavior
Memory leak should no longer be present

### Current Behavior
After 20-60 minutes almost all memory (16GB) is used

### Other details

https://github.com/user-attachments/assets/45102cb5-10dd-4d1b-af33-f0534c2fbe90

**Comment by absolutelynothelix** [comment]
does 4529db794c1f255b4c2de7acccf016b60f0a9151 fix the issue for you? note that it's in the `absolutelynothelix-buildtest` branch.

**Comment by Monsterovich** [comment]
Strange, why don't I have this issue then?

I suggest that the OP author build the picom with the `-fsanitize=address` flag so you can see where the memory leak is.

**Comment by absolutelynothelix** [comment]
closing as fixed in 4529db794c1f255b4c2de7acccf016b60f0a9151.
